[Concept Entry] R Strings #2021 (#2070)
Codecademy/docs: content/r/concepts/strings/strings.md

---
Title: 'Strings'
Description: 'Strings in R are a fundamental data type used for storing text data.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'R'
  - 'Strings'
 
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

> Strings in R are a fundamental data type used for storing text data. They are created using quotation marks, either single (') or double ("), and can contain any combination of letters, numbers, and symbols. In this entry, we'll cover the syntax for creating and manipulating strings in R.

Author: @caupolicandiaz okay got it.

## Syntax

To create a string in R, simply enclose the text you want to use in quotation marks `' '`/ `" "`. Here are some examples:

```r
# Single quotes
my_string_1 <- 'Hello, World!'

# Double quotes
my_string_2 <- "This is a string."
```

## Concatenating Strings

To combine multiple strings together, you can use the paste() function:

```r
# Concatenating two strings
string_1 <- "Hello"
string_2 <- "world"
my_string <- paste(string_1, string_2)
```
You can also use the paste0() function, which concatenates strings without any separator:

```r
# Concatenating two strings without separator
string_1 <- "Hello"
string_2 <- "world"
my_string <- paste0(string_1, string_2)
```

## Subsetting Strings

To extract a specific portion of a string, you can use square brackets `[]` and specify the starting and ending indices:
```r
my_string <- "Hello, World!"
substring <- my_string[1:5]
# Output: "Hello"

```
You can also use the substr() function to extract a substring from a string:

```r
my_string <- "Hello, World!"
substring <- substr(my_string, start = 1, stop = 5)
# Output: "Hello"

```

## Modifying Strings

> To modify a string, you can use various `functions` available in R. For example, you can convert a string to uppercase or lowercase using the `toupper()` and `tolower()` functions, respectively:

Reviewer: ```suggestion
To modify a string, you can use various functions available in R. For example, you can convert a string to uppercase or lowercase using the `toupper()` and `tolower()` functions, respectively:
```

```r
my_string <- "Hello, World!"
uppercase_string <- toupper(my_string)
# Output: "HELLO, WORLD!"

lowercase_string <- tolower(my_string)
# Output: "hello, world!"

> ```

Reviewer: add a space below

You can also remove whitespace from the beginning and/or end of a string using the `trimws()` function:

```r
my_string <- "   Hello, World!   "
trimmed_string <- trimws(my_string)
# Output: "Hello, World!"

> ```

Reviewer: add space below

> ## Example

Reviewer: ```suggestion
## Calculating the Length of a String
```

> Here's an example of using strings in R to calculate the length of a string and to subset a string:

Reviewer: ```suggestion
In R, the `nchar()` function is used to calculate the length of a string:
```

```r
# Define a string variable
my_string <- "Hello, World!"

# Calculate the length of the string
string_length <- nchar(my_string)
# Output: 13

# Extract the first 5 characters of the string
substring <- substr(my_string, start = 1, stop = 5)

> # Output: "Hello"

Reviewer: This can be deleted, it isn't really adding new information. A substring example was already addressed earlier.

---
Title: 'Strings'
Description: 'Strings in R are a fundamental data type used for storing text data.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'R'
  - 'Strings'
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

**Strings** in R are a fundamental data type used for storing text data. They are created using quotation marks, either single (') or double ("), and can contain any combination of letters, numbers, and symbols. The sections that follow will cover the syntax for creating and manipulating strings in R.

## Syntax

> To create a string in R, simply enclose the text you want to use in quotation marks `' '`/ `" "`. Here are some examples:

Reviewer 2: ```suggestion
To create a string in R, simply enclose any desired text in quotation marks `' '`/ `" "`. Here are some examples:
```

```r
# Single quotes
my_string_1 <- 'Hello, World!'

# Double quotes
my_string_2 <- "This is a string."
```

## Concatenating Strings

> To combine multiple strings together, you can use the `paste()` function:

Reviewer 2: ```suggestion
The `paste()` function combines multiple strings together:
```

```r
# Concatenating two strings
string_1 <- "Hello"
string_2 <- "world"

> my_string <- paste(string_1, string_2)

Reviewer 2: Output comment would be helpful here

```

> You can also use the `paste0()` function, which concatenates strings without any separator:

Reviewer 2: ```suggestion
The `paste0()` function concatenates strings without any separator:
```

```r
# Concatenating two strings without a separator
string_1 <- "Hello"
string_2 <- "world"

> my_string <- paste0(string_1, string_2)

Reviewer 2: here as well

```

## Subsetting Strings

> To extract a specific portion of a string, you can use square brackets `[]` and specify the starting and ending indices:

Reviewer 2: ```suggestion
Square brackets `[]`, that specify the starting and ending indices `[start:end]`, can be used to extract a specific portion of a string:
```

```r
my_string <- "Hello, World!"
substring <- my_string[1:5]
# Output: "Hello"
```

> You can also use the `substr()` function to extract a substring from a string:

Reviewer 2: ```suggestion
The `substr()` function extracts a substring from a string:
```

```r
my_string <- "Hello, World!"
substring <- substr(my_string, start = 1, stop = 5)
# Output: "Hello"
```

## Modifying Strings

> To modify a string, you can use various functions available in R. For example, you can convert a string to uppercase or lowercase using the `toupper()` and `tolower()` functions, respectively:

Reviewer 2: ```suggestion
Various functions available in R to modify a string. For example, a string can be converted to uppercase or lowercase using the `toupper()` and `tolower()` functions, respectively:
```

```r
my_string <- "Hello, World!"
uppercase_string <- toupper(my_string)
# Output: "HELLO, WORLD!"

lowercase_string <- tolower(my_string)
# Output: "hello, world!"
```

> You can also remove whitespace from the beginning and/or end of a string using the `trimws()` function:

Reviewer 2: ```suggestion
Whitespace can be removed from the beginning and/or end of a string using the `trimws()` function:
```
